unidist/core/backends/mpi/core/controller/api.py

```diff
@@ -49,6 +51,56 @@
 topology = dict()
 # The global variable is responsible for if MPI backend has already been initialized
 is_mpi_initialized = False
+threads = []
```

Comment:
Why do we need a list of threads?

Reply:
This is used to store the background thread created to a global list, a thread created in init would be required to be accessed in shutdown method to join threads. 

An alternative would be to make the thread variable global. But I thought initializing thread with None was not the right way.